lib/posix/posix_utils.nim

```diff
@@ -99,3 +99,14 @@ proc mkdtemp*(prefix: string): string =
     raise newException(OSError, $strerror(errno))
   return $tmpl
 
+when defined(amd64):
```

Comment:
So this implementation is only for amd64? Come on, we need to support it on every Posix system.

Reply:
32 Bit does work Ok, but may overflow tiny 32 Bit `uint` really easy being bytes, even if divided,
old test pass confirm that failing on 32 Bit, so I intentionally made it 64 Bit only, test pass 64 Bit.
:slightly_smiling_face:

Reply:
1. The check could look like ``when sizeof(int) == 8`` as there other 64 bit target CPUs. (But it's the wrong idea regardless.)
2. Please use the ``uint64`` type.

Reply:
@Araq Done. **Thanks for the help**, sounds totally right. :+1: